File: Sorting-Algorithms/Bubble_Sort/Cpp/bubble_sort.cpp
/*
    Bubble Sort
    Time Complexity
        Best Case Scenario  : O(n)
        Worst Case Scenario : O(n^2)
*/
#include <iostream>
using namespace std;
void swap(int *a,int *b)
{
    *a= *a + *b;
    *b= *a - *b;
    *a= *a - *b;
}
void bubble_sort(int arr[],int n)
{
    int i,j;
    for(i=0;i<n-1;i++)
    {
        for(j=0;j<n-i-1;j++)
        {
            if(arr[j]>arr[j+1])
            {
                swap(&arr[j],&arr[j+1]);
            }
        }
    }
}
int main()
{
    int arr[]={1,-10,-30,20,2,11,15};
    int n=sizeof(arr)/sizeof(arr[0]),i;
    bubble_sort(arr,n);
    cout<<"Array after bubble sort : ";
    for(i=0;i<n;i++)
    {
        cout<<arr[i]<<" ";
    }
    cout<<endl;
    return 0;
}

File: Sorting-Algorithms/Bubble_Sort/Python/bubble_sort.py
'''
    Bubble Sort
    Time Complexity
        Best Case Scenario  : O(n)
         Avergae Case Scenario : O(n^2)
        Worst Case Scenario : O(n^2)
'''
def bubble_sort(arr,n):
    for i in range(0,n-1):
        for j in range(0,n-i-1):
            if(arr[j]>arr[j+1]):
                arr[j+1],arr[j]=arr[j],arr[j+1]

arr=[1,-10,-30,20,2,11,15]
n=len(arr)
bubble_sort(arr,n)
print("Array after bubble sort : ", end='')
for i in arr:
    print("{} ".format(i),end="")

print()
